src/services/SessionEventProcessor.ts

```diff
@@ -92,5 +115,26 @@ export class SessionEventProcessor {
 			throw new AppError(HttpCodesEnum.SERVER_ERROR, error.message);
 		}
 	}
+
+	async sendOldTemplateEmail(oldSessionEvent: SessionEvent): Promise<void> {
+		// Validate all necessary fields are populated before processing the data.
+		try {
+			await this.validationHelper.validateModel(oldSessionEvent, this.logger);
```

Comment:
Haven't we already validated the model on line 72, what are we doing here that is different here?

Reply:
Will update the comments here. We are validating if all fields required to send the new template email exists to check if we can send the new template email, else fall back to the old email.